google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -216,6 +227,12 @@ private static String getValue(Label label, String resourceType) {
 
   /* Detect monitored Resource type using environment variables, else return global as default. */
   private static Resource getAutoDetectedResourceType() {
+
+    // TODO detect the function from metadata server. If not available - use env var
+    if (System.getenv("FUNCTION_NAME") != null) {
```

Comment:
I see no use of the environment variable's value. Does it intended to be used as a flag? In that case the name should follow a different convention.
Leaving the behavior in the hanging state isn't good solution. Let's return "global" until we determine the metadata server attribute.
Also, introducing a logic based on hard-coded environment variables is not a best strategy. A variable can be override by the function's logic or, if it is used as a fallback scenario, lead to a different behavior in real and simulated environments.

Reply:
The public documentation does not describe it but a quick test discover the following environment variables set (for golang environment): `K_REVISION`, `GCF_BLOCK_RUNTIME_go112`, `GCF_BLOCK_RUNTIME_nodejs6`, `K_SERVICE`, `FUNCTION_TARGET`, `FUNCTION_SIGNATURE_TYPE`. There is no `FUNCTION_NAME` environment variable.